sriovnet.go

```diff
@@ -520,3 +520,37 @@ func GetPciFromNetDevice(name string) (string, error) {
 	}
 	return base, nil
 }
+
+// GetPKeyByIndexFromPci returns the PKey stored under given index for the IB PCI device
+func GetPKeyByIndexFromPci(pciAddress string, index int) (string, error) {
+	pciDir := filepath.Join(PciSysDir, pciAddress, "infiniband")
+	dirEntries, err := utilfs.Fs.ReadDir(pciDir)
+	if err != nil {
+		return "", err
+	}
+	if len(dirEntries) == 0 {
+		return "", fmt.Errorf("infiniband directory is empty for device: %s", pciAddress)
+	}
+
+	portsDir := filepath.Join(pciDir, dirEntries[0].Name(), "ports")
+	dirEntries, err = utilfs.Fs.ReadDir(portsDir)
+	if err != nil {
+		return "", err
+	}
+	if len(dirEntries) == 0 {
+		return "", fmt.Errorf("infiniband ports directory is empty for device: %s", pciAddress)
+	}
+
+	index0FilePath := filepath.Join(portsDir, dirEntries[0].Name(), "pkeys", strconv.Itoa(index))
+	pKeyBytes, err := utilfs.Fs.ReadFile(index0FilePath)
+	if err != nil {
+		return "", err
```

Comment:
Here as well:

`        return "", fmt.Errorf("failed to read PKey file: %v", err)
`